Added Deployment Strategy example in sample deployment (#42598)
kubernetes/website: content/en/docs/concepts/workloads/controllers/deployment.md

  * The Pod template's specification, or `.template.spec` field, indicates that
  the Pods run one container, `nginx`, which runs the `nginx`
  [Docker Hub](https://hub.docker.com/) image at version 1.14.2.
  * Create one container and name it `nginx` using the `.spec.template.spec.containers[0].name` field.  
  
* The `strategy` field specifies the strategy used to replace old Pods by new ones.

>   The `.strategy.type` can be "Recreate" or "RollingUpdate". "RollingUpdate" is the default value.

Reviewer: ```suggestion
  For a Deployment, `.spec.strategy.type` can be `RollingUpdate` (the default) or `Recreate`.
```

>   `strategy.rollingUpdate.maxUnavailable` and `strategy.rollingUpdate.maxSurge` are optional fields
>   which specifies the maximum number of Pods that can be unavailable during the update process
>   and the maximum number of Pods that can be created over the desired number of Pods respectively.

Reviewer: (nit)
```suggestion
  `.spec.strategy.rollingUpdate.maxUnavailable` and `.spec.strategy.rollingUpdate.maxSurge`
  are optional fields;
  `maxUnavailable` lets you specify the maximum number of Pods that can be
  unavailable during the rolling update, and `maxSurge` lets you set a maximum number of
  additional Pods that can be created for the duration of the rollout.
```

  * The Pod template's specification, or `.template.spec` field, indicates that
  the Pods run one container, `nginx`, which runs the `nginx`
  [Docker Hub](https://hub.docker.com/) image at version 1.14.2.
  * Create one container and name it `nginx` using the `.spec.template.spec.containers[0].name` field.  
  

> * The `strategy` field specifies the strategy used to replace old Pods by new ones. The `.strategy.type` can be "Recreate" or "RollingUpdate". "RollingUpdate" is the default value. `strategy.rollingUpdate.maxUnavailable` and `strategy.rollingUpdate.maxSurge` are optional fields which specifies the maximum number of Pods that can be unavailable during the update process and the maximum number of Pods that can be created over the desired number of Pods respectively.

Reviewer 2: ```suggestion
* The `strategy` field specifies the strategy used to replace old Pods by new ones.
  The `.strategy.type` can be "Recreate" or "RollingUpdate". "RollingUpdate" is the default value.
  `strategy.rollingUpdate.maxUnavailable` and `strategy.rollingUpdate.maxSurge` are optional fields
  which specifies the maximum number of Pods that can be unavailable during the update process
  and the maximum number of Pods that can be created over the desired number of Pods respectively.
```
Provide some newline chars for easy review.
